src/shared/conf-parser.c

```diff
@@ -502,19 +504,16 @@ static int config_parse_many_files(
         }
 
         STRV_FOREACH(fn, files) {
-                _cleanup_free_ struct stat *st_dropin = NULL;
                 _cleanup_fclose_ FILE *f = NULL;
-                int fd;
-
-                f = fopen(*fn, "re");
-                if (!f) {
-                        if (errno == ENOENT)
-                                continue;
+                _cleanup_free_ char *fname = NULL;
 
-                        return -errno;
-                }
+                r = chase_and_fopen_unlocked(*fn, root, CHASE_AT_RESOLVE_IN_ROOT, "re", &fname, &f);
+                if (r == -ENOENT)
+                        continue;
+                if (r < 0)
+                        return r;
 
-                fd = fileno(f);
+                int fd = fileno(f);
 
                 r = ordered_hashmap_ensure_put(&dropins, &config_file_hash_ops_fclose, *fn, f);
```

Comment:
`*fn` -> `fname`?

Reply:
No, for two reasons:
- `fname` is freed when we go out of the block, so we would need to change the lifetime if it was to be used as key.
- It's actually OK to use `*fn` as the key. Even if the same file happened to be linked twice under different names with different priority, we would still want to read it twice in the specified order.